### PR TITLE
Switch from Flow to Typescript

### LINKED ISSUE #5: Migrate from flow to typescript
I originally chose flow over typescript because it allows putting type definitions in comments and as such doesn't require a separate compilation step. Unfortunately flow has several limitations including not supporting bigints and some incorrect builtin type declarations. As such I think using typescript is better in the long term.

### PR DESCRIPTION
**Pull request by Systemcluster**
This is a follow-up to #15 addressing the comments in https://github.com/bjorn3/browser_wasi_shim/pull/15#issuecomment-1445115620, rebased on current main. I also added a [npm prepare step](https://docs.npmjs.com/cli/v9/using-npm/scripts#life-cycle-scripts) to build the types and JS files for the published package.

Additionally I added a GitHub workflow for running checks for PRs and pushes; if that's not desired I'll remove it again.

Closes #15 
Closes #5

### CODE REVIEW

**Comment by Systemcluster** [comment]
There is one issue that jumped out when trying to use the library now with full typing:

The examples use `new File()` for the `fds` array when constructing `WASI`, but `File` does not implement `Fd`. The types here were taken directly from the flow typing, and it looks like `File` doesn't implement the necessary methods, so I assume `OpenFile` is meant to be used instead. (If so, the docs and examples should be updated.)

Another detail that might be improvable is that there are internally a lot of inconsistencies between `BigInt` and `Number` that are ignored with `@ts-ignore` directives, it might make sense to clear that up in a separate PR.

File: .npmignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ node_modules
 *.wasm.gz
 /examples
 .github/
+src/
+.flowconfig
```
Comment (bjorn3) [inline .npmignore:7]:
```suggestion
```
This file is now removed.

**Comment by bjorn3** [comment]
Thanks @A-K-O-R-A for doing the initial port and @Systemcluster for fixing my review comments.

**Comment by bjorn3** [comment]
Released as v0.2.5.

**Comment by bhelx** [comment]
This is great, thanks for picking it up @Systemcluster !

**Pull request merged**